pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -0,0 +1,129 @@
+package autosizednodes
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/ignition/v2/config/util"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+type Reconciler struct {
+	client.Client
+	Scheme *kruntime.Scheme
+
+	log    *logrus.Entry
+	arocli aroclient.Interface
+}
+
+const (
+	configName = "dynamic-node"
+)
+
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, mgr ctrl.Manager) *Reconciler {
+
+	return &Reconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+
```

Comment:
Client and Scheme are used internally via calling `r.Get` which uses the internal client with cache enabled. I decided to explicitly create create client so I am 100% is the same manager client as for the rest.